packages/x-ui-collections-bundle/src/graphql/Collection.ts

```diff
@@ -893,4 +910,35 @@ export abstract class Collection<T = null> {
   protected isUpdateModifier(element: any): element is UpdateFilter<T> {
     return element["$set"];
   }
+
+  /**
+   * Provide a Apollo.OptimisticResponse object based on the provided _id and document
+   */
+  public optimistic(
+    _id: ObjectId | string,
+    document: Partial<T>,
+    operationName?: string
+  ) {
+    const stringId = _id instanceof ObjectId ? _id.toString() : _id;
+
+    // We can't be optimistic if we do not know the operationName
+    if (!operationName) return undefined;
+
+    const collectionName = this.getName?.();
+
+    // We can't be optimistic if we do not know the __typename
+    if (!collectionName) return undefined;
+
+    return {
+      // `${this.getName()}UpdateOne`
+      [operationName]: {
+        _id: stringId,
+        __typename:
+          collectionName[collectionName.length - 1] === "s"
```

Comment:
This is not a reliable way to do this, we might need to add a getTypeName() { }

Reply:
Moved it to collection.getTypeName, keeping this dumb implementation. Probably that the best would be to have the generator to auto implement an override of `getTypename` and jut keep this dumb implementation "just in case" ?